react-ui/src/pages/contest/ContestGrid.jsx

```diff
@@ -0,0 +1,255 @@
+  const { data: contest } = useSwrData(apiPath, false);
+
+  const classes = useStyles();
+  const [{ votingDisabled }, setComponentsState] = useComponentsState();
+  const { gridDisplayWidth, winnerDisplayWidth } = useContestSizing();
+
+  const [{ density = 'default' }] = useSettingsState();
+
+  const votingUnavailable = votingDisabled || votingExpired;
+
+  const getGridVariables = useCallback((fullWidth) => {
+    const xs = 12;
+    let sm = 12;
+    let md = 6;
+    let lg = 4;
+
+    if (fullWidth) {
+      md = 12;
+      lg = 12;
+    } else if (density === 'compact') {
+      sm = 6;
+      md = 4;
+      lg = 3;
+    }
+
+    return {
+      xs,
+      sm,
+      md,
+      lg,
+    };
+  }, [density]);
+
+  const {
+    categories,
+    entries,
+    isContestMode,
+  } = contest;
+
+  return (
+    <Grid container spacing={density === 'compact' ? 1 : 2}>
+      {entries
+        .filter(
+          // eslint-disable-next-line max-len
+          ({ category }) => !selectedCategories.length || selectedCategories.includes(category),
+        )
+        .map(
+          ({
+            average,
+            category,
+            categoryRank,
+            id,
+            imagePath,
+            imgurId,
+            height,
+            name: entryName,
+            rank,
+            rating,
+            user,
+            width,
+          }) => (
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            <Grid key={id} item {...getGridVariables(rank === '1')}>
+              <Card id={id} className={classes.entry}>
+                <CardContent className={classes.entryHeading}>
+                  {rank && (
+                    <Typography component="div" variant="h6">
+                      <span className={classes.numberSymbol}>#</span>
+                      {rank}
+                    </Typography>
+                  )}
+                  <div className={classes.entryInfo}>
+                    <Box alignItems="flex-start" display="flex" flexGrow={1}>
+                      <Box flexGrow={1}>
+                        <Typography component="div" variant="subtitle2">
+                          {entryName}
+                        </Typography>
+                        {user && (
+                          <Typography variant="caption">
+                            <RedditUserAttribution user={user} />
+                          </Typography>
+                        )}
+                      </Box>
+                      {/* <Experiment name="contest_card_description">
+                        <CustomIconButton
+                          ariaLabel="View description"
+                          className={classes.descriptionIcon}
+                          Icon={DescriptionIcon}
+                          onClick={() => {
+                            viewDescription(id);
+                          }}
+                          size="small"
+                        />
+                      </Experiment> */}
```

Comment:
Why is this commented out?

Reply:
So the logic didn't fit super well because viewDescription calls both `setDescriptionEntryId(entryId)` and `setSettingsOpen(true)`.

If you want to keep this active, I recommend `useMemo(() => {}, [entryId])` to render this guy. I'm also not a big fan of keeping unfinished code in the prod, so I might also gate it with an `isDev`. Let me know which you prefer.

Reply:
The `<Experiment>` component is functionally equivalent to gating with `isDev`, since they both won't render the content unless it's required.

I'm not getting what the issue is with the logic. What problem does `useMemo` solve? I'm not very familiar with this hook.

Reply:
`useMemo` is the variable / component equivalent of `useCallback`. It is primarily used for limited re-renders of computational intensive components and can optimize passing a component as a prop, but I was primarily suggesting it to bring the experimental logic out of the return state and raise it somewhere more discrete. 

I've left it as-is though.